lib/agent/index.js

```diff
@@ -122,15 +137,25 @@ class Agent extends EventEmitter {
     for (const pool of this[kClients].values()) {
       closePromises.push(pool.close())
     }
-    return Promise.all(closePromises)
+    return Promise.all(closePromises).then((closedPromises) => {
+      return new Promise((resolve) => {
+        this[kClosed] = true
+        resolve(closedPromises)
+      })
+    })
   }
 
   destroy () {
     const destroyPromises = []
     for (const pool of this[kClients].values()) {
       destroyPromises.push(pool.destroy())
     }
-    return Promise.all(destroyPromises)
+    return Promise.all(destroyPromises).then((destroyedPromises) => {
+      return new Promise((resolve) => {
+        this[kDestroyed] = true
```

Comment:
This should be set directly, please see how it's done in `Pool` and `Client`. Also should set `this[kClosed]`.